docs/peer_forwarder.md

```diff
@@ -0,0 +1,123 @@
+* `awsCloudMapNamespaceName`(Optional) - A `String` representing the CloudMap namespace when using AWS CloudMap service discovery. Required if `discovery_mode` is set to `aws_cloud_map`.
+* `awsCloudMapServiceName`(Optional) - A `String` representing the CloudMap service when using AWS CloudMap service discovery. Required if `discovery_mode` is set to `aws_cloud_map`.
+* `aws_cloud_map_query_parameters`(Optional): A `Map` of Key/value pairs to filter the results based on the custom attributes attached to an instance. Only instances that match all the specified key-value pairs are returned.
+* `buffer_size`(Optional): An `int` representing max number of unchecked records the buffer accepts (num of unchecked records = num of records written into the buffer + num of in-flight records not yet checked by the Checkpointing API). Default is `512`.
+* `batch_size`(Optional): An `int` representing max number of records the buffer returns on read. Default is `48`.
+
+### SSL
+The SSL configuration for setting up trust manager for peer forwarding client to connect to other Data Prepper instances.
+
+* `ssl`(Optional) : A `boolean` that enables TLS/SSL. Default value is `false`.
+* `ssl_certificate_file`(Optional) : A `String` representing the SSL certificate chain file path or AWS S3 path. S3 path example `s3://<bucketName>/<path>`. Required if `ssl` is set to `true`.
+* `ssl_key_file`(Optional) : A `String` represents the SSL key file path or AWS S3 path. S3 path example `s3://<bucketName>/<path>`. Required if `ssl` is set to `true`.
+* `ssl_insecure_disable_verification`(Optional) : A `boolean` that disables the verification of server's TLS certificate chain. Default value is `false`.
+* `use_acm_certificate_for_ssl`(Optional) : A `boolean` that enables TLS/SSL using certificate and private key from AWS Certificate Manager (ACM). Default is `false`.
+* `acm_certificate_arn`(Optional) : A `String` represents the ACM certificate ARN. ACM certificate take preference over S3 or local file system certificate. Required if `use_acm_certificate_for_ssl` is set to `true`.
+* `acm_certificate_timeout_millis`(Optional) : An 'int' representing the timeout in milliseconds for ACM to get certificates. Default value is `120000`.
+* `aws_region`(Optional) : A `String` represents the AWS region to use `ACM`, `S3` or `AWS Cloudmap`. Required if `use_acm_certificate_for_ssl` is set to `true` or `ssl_certificate_file` and `ssl_key_file` is `AWS S3` path or if `discovery_mode` is set to `aws_cloud_map`.
+
+```yaml
+peer_forwarder:
+  ssl: true
+  ssl_certificate_file: "<cert-file-path>"
+  ssl_key_file: "<private-key-file-path>"
+```
+
+### Authentication
+
+* `authentication`(Optional) : A `Map` that enables mTLS. It can either be `mutual_tls` or `unauthenticated`. Default value is `unauthenticated`.
+```yaml
+peer_forwarder:
+  authentication:
+    mutual_tls:
+```
+
+### <a name="DNS_Discovery"></a>Operating a Cluster with DNS Discovery
```

Comment:
It might be good to expand on what this should be. I believe this is typically an `A` record which is populated with a list of IP addresses.